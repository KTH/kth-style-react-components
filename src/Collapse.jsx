import React from 'react'
import { Collapse as BootstrapCollapse } from 'reactstrap'
import classNames from 'classnames'

class Collapse extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      isOpen: false,
      firstLoad: 'true'
    }

    this.toggleHeader = this.toggleHeader.bind(this)
  }

  toggleHeader () {
    this.setState(state => ({
      isOpen: !state.isOpen,
      firstLoad: 'false'
    }))
  }

  render() {
    const classes = classNames(
      'card',
      'collapsible',
      {'white': this.props.color === 'white'},
      {'blue': (this.props.color === 'blue' || !this.props.color)}, // Default color is blue
    )

    return <div className={classes}>
      <div class="card-header" role="tab" >
        <a onClick={this.toggleHeader} aria-expanded={this.state.isOpen} load={this.state.firstLoad}>{this.props.buttonText}</a>
      </div>
      <BootstrapCollapse isOpen={this.state.isOpen}>
        <div class='card-body'>
          {this.props.children}            
        </div>
      </BootstrapCollapse>
    </div>        
  }
}

export default Collapse