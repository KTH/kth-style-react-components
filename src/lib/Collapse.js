import React from 'react'
import { Collapse as BootstrapCollapse } from 'reactstrap'
import classNames from 'classnames'

class Collapse extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      isOpen: false,
      firstLoad: 'true'
    }

    this.toggleHeader = this.toggleHeader.bind(this)
  }

  toggleHeader() {
    this.setState(state => ({
      isOpen: !state.isOpen,
      firstLoad: 'false'
    }))
  }

  render() {
    const classes = classNames(
      'card',
      'collapsible',
      { white: this.props.color === 'white' },
      { blue: this.props.color === 'blue' || !this.props.color } // Default color is blue
    )

    return (
      <div className={classes}>
        <div className="card-header" role="tab">
          <h4 className="mb-0">
            <a
              href="#"
              onClick={this.toggleHeader}
              aria-expanded={this.state.isOpen}
              load={this.state.firstLoad}
            >
              {this.props.buttonText}
            </a>
          </h4>
        </div>
        <BootstrapCollapse isOpen={this.state.isOpen}>
          <div className="card-body">{this.props.children}</div>
        </BootstrapCollapse>
      </div>
    )
  }
}

export default Collapse
