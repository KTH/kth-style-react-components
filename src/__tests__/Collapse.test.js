import React from "react";
import renderer from "react-test-renderer";
import Collapse from '../lib/Collapse';

const DEFAULT_COLOR = {
  buttonText: 'Click me',
  content: "This is a test string that should render inside collapse component"
}

const WHITE_COLOR = {
  color: 'white',
  buttonText: 'Click me',
  content: "This is a test string that should render inside collapse component"
}

describe("Collapse", () => {
  it("default color should be blue", () => {
    const tree = renderer
      .create(<Collapse buttonText={DEFAULT_COLOR.buttonText}>
        <p>{DEFAULT_COLOR.content}</p>
      </Collapse>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be white when given props color=white", () => {
    const tree = renderer
      .create(<Collapse color={WHITE_COLOR.color} buttonText={WHITE_COLOR.buttonText}>
        <p>{WHITE_COLOR.content}</p>
      </Collapse>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});