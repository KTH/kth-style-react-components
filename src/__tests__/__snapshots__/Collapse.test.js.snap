// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse default color should be blue 1`] = `
<div
  className="card collapsible blue"
>
  <div
    className="card-header"
    role="tab"
  >
    <h4
      className="mb-0"
    >
      <a
        aria-expanded={false}
        href="#"
        load="true"
        onClick={[Function]}
      >
        Click me
      </a>
    </h4>
  </div>
  <div
    aria-expanded="false"
    className="collapse"
    style={Object {}}
  >
    <div
      className="card-body"
    >
      <p>
        This is a test string that should render inside collapse component
      </p>
    </div>
  </div>
</div>
`;

exports[`Collapse should be white when given props color=white 1`] = `
<div
  className="card collapsible white"
>
  <div
    className="card-header"
    role="tab"
  >
    <h4
      className="mb-0"
    >
      <a
        aria-expanded={false}
        href="#"
        load="true"
        onClick={[Function]}
      >
        Click me
      </a>
    </h4>
  </div>
  <div
    aria-expanded="false"
    className="collapse"
    style={Object {}}
  >
    <div
      className="card-body"
    >
      <p>
        This is a test string that should render inside collapse component
      </p>
    </div>
  </div>
</div>
`;
